contrib/win32/win32compat/termio.c

```diff
@@ -257,7 +254,7 @@ syncio_close(struct w32_io* pio)
 	/* If io is pending, let worker threads exit. */
 	if (pio->read_details.pending) {
 		/* For console - the read thread is blocked so terminate it. */
-		if (FILETYPE(pio) == FILE_TYPE_CHAR && in_raw_mode)
+		if (FILETYPE(pio) == FILE_TYPE_CHAR && (!IsWindows8OrGreater() || in_raw_mode))
```

Comment:
recommend adding a MACRO like this for readability
#define IsWin7OrLess()  (!IsWindows8OrGreater())